assets/prototype/domain/eventEditor/containers/queries/useDatetimeIds.js

```diff
@@ -0,0 +1,31 @@
+import * as R from 'ramda';
+import { useApolloClient } from '@apollo/react-hooks';
+import { GET_DATETIMES } from './dates';
+
+const useDatetimeIds = () => {
+	const eventId = R.pathOr(null, ['eeEditorEventData', 'eventId'], window);
+	const client = useApolloClient();
+	let data;
+
+	try {
+		data = client.readQuery({
+			query: GET_DATETIMES,
+			variables: {
+				where: {
+					eventId,
+				},
+			},
+		});
+	} catch (error) {
+		throw new Error(error);
+	}
+
+	const getDatetimes = (data) => R.pathOr([], ['datetimes', 'nodes'], data);
+	const getIds = (datetimes) => R.map(({ id }) => id, datetimes);
+	const getDatetimeIds = R.pipe(getDatetimes, getIds);
+	const datetimeIds = getDatetimeIds(data);
+
+	return datetimeIds;
```

Comment:
Why not simply this?
```js
return R.pathOr([], ['datetimes', 'nodes'], data).map(({ id }) => id);
```
Also, `data` in `getDatetimes()` is already defined in the upper scope.

Reply:
That is a great question.
When error will be thrown we will have `data = {}`
and in this case the `.map(({ id }) => id)` will throw an error saying that there is no `id` property, because it will try to map over an empty array.
This is why I've updated this:
`R.map((datetime) => R.prop('id', datetime), datetimes)`
in this case when we will have empty array `R.prop('id', datetime)` will result in undefined, not with an unexpecte error.

Reply:
Even if `data` is `{}`, `R.pathOr([], ['datetimes', 'nodes'], data)` should take care of that and return an empty array.

Reply:
![image](https://user-images.githubusercontent.com/18226415/70132933-31444d80-16ab-11ea-88ab-907864d56a8e.png)